product_customer_code_picking/models/stock_move.py

```diff
@@ -0,0 +1,26 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013-2017 Agile Business Group sagl
+#     (<http://www.agilebg.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, fields, models
+
+
+class StockMove(models.Model):
+    _inherit = 'stock.move'
+
+    @api.multi
+    @api.depends('picking_id.partner_id', 'product_id',
+                 'product_id.customer_ids')
+    def _compute_product_customer_code(self):
+        for move in self.filtered(lambda m: m.picking_id and
+                                  m.picking_id.partner_id and
+                                  m.product_tmpl_id.customer_ids):
+            move.product_customer_code = \
+                move.product_tmpl_id.customers_ids.filtered(
```

Comment:
I think the filtered should be on the move recordset (This generates error). But eventually it is no more relevant as the condition is done at the loop level.

Moreover, don't be afraid to depends on 'product_id.customer_ids.product_code'